Batch data into database

phipsae opened the pull request:
Hi Carlos,

and now follows the PR for the Batch Data :) 
I first created the data structure with the respective overview page, create und update functionality.
At the moment, no batch data is linked to other components, like Batch Join Banner. That is planned for the next PR. 
First I wanted to make sure that this works and get your feedback, as this is already a quite big PR. 
I tried my best to find all errors.
See for functionality the quick video. 


https://github.com/user-attachments/assets/54d9cfb7-57d4-458d-b1bb-a81821b3a02e

Thanks!

carletex commented on packages/react-app/data/socials.js:
Why are we mixing the builder socials with the batch links?

Would this show when a builder try to update their socials?

I think we should have a separate context/file/variable for it.

carletex reviewed (commented):
Thanks for this @phipsae !! Excited about this one.

Haven't tested it yet, but left a few comments (some of them based on our previous conversations about being more flexible). Hopefully we can make this as simple as possible.

Let me know if you have any questions, or if something I said doesn't make sense to you. Also happy to get together!

> At the moment, no batch data is linked to other components, like Batch Join Banner. That is planned for the next PR.
> First I wanted to make sure that this works and get your feedback, as this is already a quite big PR.

Great call! Make sense to split the PRs

Thanks!!

phipsae commented on packages/backend/routes/builders.js:
```diff
@@ -14,12 +14,24 @@ router.get("/", async (req, res) => {
 
-router.get("/batches", async (req, res) => {
-  console.log("/builders/batches");
+router.get("/batchBuilders", async (req, res) => {
```
No, I thought it might make sense to have it the same name like the page file. But change it back.

phipsae commented on packages/backend/services/db/db.js:
```diff
@@ -232,6 +242,12 @@ module.exports = {
 
+  findAllBatches,
+  findBatchByNumber,
+  findBatchById,
```
for editing reasons

phipsae commented on packages/backend/scripts/20241016_create_batch_data.js:
```diff
@@ -0,0 +1,134 @@
+// // Docs: https://firebase.google.com/docs/firestore/quickstart#node.js_1
+const db = firebaseAdmin.firestore();
+
+// Missing telegram links up to batch 7 - link to dummy tg group
```
Added yours, asked Zak, as soon as I have them I updated. 
But we could do this also live, as soon as we merge.

phipsae commented:
Hi Carlos,

I've implemented all the suggested changes. Regarding the Telegram links, I wasn't sure whether to leave them in since they were already posted here, so I kept them and updated them with your links. I'll update the links from Austin or Zak as soon as I receive them, but we can also handle that after the PR goes live.

Thanks!

carletex reviewed (approved):
Great job @phipsae !! I pushed some little things.

I hope that we are not missing anything since this was a big PR.

phipsae commented:
Thank you! I hope so, too 😅but I think everything should work :)
